hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/query/QueryWorkflowImpl.java

```diff
@@ -171,7 +173,11 @@ public void handleQuery(@NonNull final Bytes requestBuffer, @NonNull final Buffe
             }
 
             // 3. Check query throttles
-            if (throttleAccumulator.shouldThrottleQuery(function, query)) {
+            if (synchronizedThrottleAccumulator.shouldThrottle(function, query)
+                    && !RESTRICTED_FUNCTIONALITIES.contains(function)) {
+                if (!responseType.equals(COST_ANSWER)) {
+                    throw new PreCheckException(NOT_SUPPORTED);
```

Comment:
This comes as a surprise. Why are we returning `NOT_SUPPORTED` here?

Reply:
`UnsupportedQueriesRegression.verifyUnsupportedOps` is failing with `expected: <NOT_SUPPORTED> but was: <BUSY>!`

Debugging the mono implementation - the throttle is returning back **BUSY** response code then we go through `GetLiveHashAnswer.responseGiven` and we return **NOT_SUPPORTED** instead of **BUSY** if `type != COST_ANSWER`. But now I notice that we are returning **NOT_SUPPORTED** if the `type == COST_ANSWER` as well 🤔 

This is happening when we call **hygieneCheck** in `StakedAnswerFlow.satisfyUsing` and if the `hygieneCheck != OK`. Maybe this is some kind of a pre-check and I should move that logic there? Not sure, any idea? I will keep looking at that.

Reply:
Okay, I deleted my previous comment since it's confusing.

I think I understand the problem now. The immediate fix I can think of is to move the throttling after the validations. You guys can tell me if this is a problem but I can't think of a different solution.

If the throttling is before the validation - in the `shouldThrottle` method we try to get the ThrottleReqsManager but since these operations(listed in `UnsupportedQueriesRegression.verifyUnsupportedOps`) are not supported we don't have _managers_ and we return BUSY.

Any concerns about this change?

Reply:
The other option would be to throw `NOT_SUPPORTED` inside the `shouldThrottle` method if the _manager_ is not found.